doc/source/development.rst

```diff
@@ -193,143 +201,242 @@ Make and commit your changes.
+To prepare Git to sign commits, follow these instructions:
 
-1. Create a key suitable for signing (its not recommended to use
-   existing keys to not mix it up with your email environment etc):
+#. Create a key suitable for signing (it is not recommended to use
+   existing keys to not mix it with your email environment):
 
-   .. code:: bash
+   .. code-block:: shell-session
 
-    $ gpg --gen-key
+    $ gpg2 --expert --full-gen-key
 
-2. Choose a DSA key (3) with a key size of 2048 bits (default) and a
-   validation of 3 years (3y). Enter your name/email and GPG will
-   generate a DSA key for you.
+#. Either choose a RSA key for signing (option `(4)`) or an ECC key for
+   signing (option `(10)`). For a RSA key choose a key size of 4096 bits
+   and for a ECC key choose Curve 25519 (option `(1)`). Enter a reasonable
+   validity period (we recommend 2 to 5 years). Complete the key generation
+   by entering your name and email address.
 
-   You can also choose to use an empty passphrase, despite GPG's warning,
-   because you are only going to sign your public git commits with it and
-   don't need it for protecting any of your secrets. That might ease later
-   use if you are not using an :command:`gpg-agent` that caches your passphrase
-   between multiple signed Git commits.
 
-3. Add the key ID to your git config
+#. Add the key ID to your git configuration
 
-   In above case, the ID is 11223344 so you add it to either your global
-   :file:`~/.gitconfig` or even better to your :file:`.git/config`
-   inside your repo:
+   Add the following entries to :file:`~/.gitconfig` or :file:`.git/config`
+   inside your repository (the latter will only apply to this repository)
+   substituting the `11223344` with your signkey's ID:
 
    .. code:: ini
 
-    [user]
-    name = Joe Developer
-    email = developer@foo.bar
-    signingkey = 11223344
+      [user]
+              name = Joe Developer
+              email = developer@foo.bar
+              signingkey = 11223344
+
+      [commit]
+              gpgsign = true
+
+   Or apply these settings via :command:`git config`:
+
+   .. code-block:: shell-session
+
+      $ git config --local user.signingkey $YOUR_SIGN_KEY_ID
+      $ git config --local commit.gpgSign true
+
+   Omitting the flag `--local` will add these settings to
+   :file:`~/.gitconfig` and make them global.
 
-Raising Versions
-~~~~~~~~~~~~~~~~
 
-The KIWI project follows the `Semantic Versioning <https://semver.org>`__
-method. To make it easier to follow this method, :command:`bumpversion` is
-used as a tool.
+Bumping the Version
+~~~~~~~~~~~~~~~~~~~
+
+The KIWI project follows the `Semantic Versioning <https://semver.org>`_
+scheme. We use the :command:`bumpversion` tool for consistent versioning.
 
-Follow these instructions to raise the major, minor, or patch part of a
-version:
+Follow these instructions to bump the major, minor, or patch part of the
+KIWI version:
```

Comment:
Before you run the `bumpversion` command, I would recommend to make sure you have a clean repo (= no modified or unknown files). If the repo is not clean, the command complains with an awful traceback.